utils/retrieval_evaluator.py

```diff
@@ -0,0 +1,559 @@
+            input_memories = [m['query'] for m in matched_input.get('memories', [])]
+
+            matched_relevant = sum(1 for query in input_memories if any(self.fuzzy_match(query, true_query) for true_query in true_relevant_queries))
+            expected_relevant = len(true_relevant_queries)
+            irrelevant = sum(1 for query in input_memories if not any(self.fuzzy_match(query, true_query) for true_query in true_relevant_queries) and not self.is_trivia_memory({'query': query}))
+            trivia = sum(1 for query in input_memories if self.is_trivia_memory({'query': query}))
+
+            retrieved_count = matched_input.get('retrieved_memories_count', len(input_memories))
+            filtered_count = len(input_memories)
+
+            total_true_relevant += matched_relevant
+            total_expected_relevant += expected_relevant
+            total_irrelevant += irrelevant
+            total_trivia += trivia
+            total_retrieved += retrieved_count
+            total_filtered += filtered_count
+
+            results.append({
+                'query': ref_entry['query'],
+                'true_relevant': matched_relevant,
+                'expected_relevant': expected_relevant,
+                'true_relevant_percent': (matched_relevant / expected_relevant * 100) if expected_relevant > 0 else 0,
+                'irrelevant': irrelevant,
+                'trivia': trivia,
+                'total_memories': filtered_count,
+                'test': ref_entry['test'],
+                'is_scored_question': ref_entry['is_scored_question'],
+                'retrieved_memories_count': retrieved_count,
+                'filtered_memories_count': filtered_count
+            })
+
+        important_entries = len(results)
+        recall_score = (total_true_relevant / total_expected_relevant) if total_expected_relevant > 0 else 0
+
+        summary = {
+            'recall_score': recall_score,
+            'total_true_relevant_percent': (total_true_relevant / total_expected_relevant * 100) if total_expected_relevant > 0 else 0,
+            'irrelevant_memories_percent': (total_irrelevant / total_filtered * 100) if total_filtered > 0 else 0,
+            'total_trivia_memories_percent': (total_trivia / total_filtered * 100) if total_filtered > 0 else 0,
+            'important_entries': important_entries,
+            'total_retrieved': total_retrieved,
+            'total_filtered': total_filtered
+        }
+
+        return results, summary
+
+    def plot_original_results(self, results, summary, output_path):
+        df = pd.DataFrame(results)
+        df['Entry'] = range(1, len(df) + 1)
+
+        plt.style.use('default')
+        fig = plt.figure(figsize=(24, 16))
+        gs = GridSpec(3, 2, height_ratios=[1, 0.5, 3], width_ratios=[1.2, 0.8])
+
+        colors = {'True_relevant': '#4CAF50', 'Irrelevant': '#FFC107', 'Trivia': '#9C27B0', 'Missing_Relevant': '#F44336'}
```

Comment:
You've used `True_**r**elevant` and `Missing_**R**elevant`, and in other places I've seen you accessing the dict with `colors[category.capitalize()]`. This would work in the first case but not in the second.